Firestore/core/src/api/load_bundle_task.h

```diff
@@ -153,13 +154,27 @@ class LoadBundleTask {
       : user_executor_(std::move(user_executor)) {
   }
 
+  ~LoadBundleTask();
+
+  /**
+   * Instructs the task to notify the specified observer when there is a
+   * progress update.
+   *
+   * @return A handle that can be used to remove the callback from this task.
+   */
+  LoadBundleHandle Observe(ProgressObserver observer);
+
   /**
    * Instructs the task to notify the specified observer when there is a
    * progress update.
    *
+   * For a given progress update, this observer is guaranteed to be called
+   * after all other observers. Calling `ObserveAtLast` a second time will
+   * override the observer registered the first time.
+   *
    * @return A handle that can be used to remove the callback from this task.
    */
-  LoadBundleHandle Observe(ProgressObserver callback);
+  LoadBundleHandle ObserveAtLast(ProgressObserver observer);
```

Comment:
Consider a name like `SetLastObserver`. It won't be symmetrical with `Observe`, but since this function has different semantics (only one last observer can be added), this seems okay.

Reply:
Done.